torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -33,13 +33,20 @@ std::vector<at::Tensor> ProcessGroup::Work::result() const {
 
 void ProcessGroup::Work::synchronize() {}
 
-void ProcessGroup::Work::wait() {
+bool ProcessGroup::Work::wait() {
   std::unique_lock<std::mutex> lock(mutex_);
   cv_.wait(lock, [&] { return completed_; });
   if (exception_) {
     std::rethrow_exception(exception_);
   }
   synchronize();
+  return !aborted_;
+}
+
+void ProcessGroup::Work::abort() {
```

Comment:
Just to confirm, we can only abort for `ProcessGroupGloo`, correct? Adding this here would make this default implementation available to `ProcessGroupNCCL` and `ProcessGroupMPI`, and since it doesn't do anything other than flip a boolean flag, I wonder if we should add a comment mentioning this?

Reply:
Correct. I only added this capability to ProcessGroupGloo to quickly unblock us. 

For  NCCL Work and MPI AsyncWork, I always returns true for wait().

Only MPI `WorkMPI` is using ProceessGroup::Work base implementation.